src/editor.ts

```diff
@@ -1,265 +1,272 @@
+      'default': string
+    };
+
+    type modeType = {
+      deploy: languagePreferenceType;
+      beta: languagePreferenceType;
+      debug: languagePreferenceType;
+    };
+
+    // Set up the defaults for language and mode
+
+    const defaultPreference: keyof languagePreferenceType = 'en';
+    let preference: keyof languagePreferenceType = defaultPreference;
+    let mode: keyof modeType = 'deploy';
+
+    // URLS by language and mode
+
+    const defaultUrls: languagePreferenceType = {
+      en: 'https://matt.engageLively.com/users/rick/published/studio-en/index.html?',
+      ja_JP:
+        'https://matt.engageLively.com/users/rick/published/studio-jp/index.html?',
+      'default':
+        'https://matt.engageLively.com/users/rick/published/studio-en/index.html?'
+    };
+
+    const urls: modeType = {
+      deploy: defaultUrls,
+      beta: defaultUrls,
+      debug: defaultUrls
+    };
+
+    // read any set values
+
+    if (galyleoSettings && debugMode) {
+      if (galyleoSettings.get('mode').composite as string) {
+        mode = galyleoSettings.get('mode').composite as keyof modeType;
+      }
+    }
+
+    if (languagePreference) {
+      preference = languagePreference.get('locale')
+        .composite as keyof languagePreferenceType;
+    }
+    if (preference === undefined) {
+      preference = defaultPreference;
+    }
+    return urls[mode][preference];
+  }
+
+  async _render(): Promise<void> {
+    // now set the src accordingly on the iframe....?
+    const filePath = this._context.path;
+    // const sessionId = this._context.model.session;
+    // dig out the user from the URL; it will be the component of the path with an @ in it
+    const parentUrl = window.location.href;
+    const components = parentUrl.split('/');
+    const userComponents = components.filter(comp => comp.indexOf('@') >= 0);
+    const user = userComponents.length > 0 ? userComponents[0] : '';
+    // assemble the url and load it into the iframe
+    // const baseURL =
+    //   'https://matt.engagelively.com/users/rick/published/Dashboard%20Studio%20Development/index.html?';
+    // for debug:
+    // const baseURL =
+    //   'https://matt.engagelively.com/worlds/load?name=Dashboard%20Studio%20Development&';
+    // both of these are defined in constants.ts, NOT versioned to avoid bogus changes and stashes.
+    const baseURL = await this._baseUrl();
+    // this._iframe.src = `${baseURL}dashboard_file=${filePath}&session=${sessionId}&inJupyterLab=true&user=${user}`;
+    this._iframe.src = `${baseURL}dashboard_file=${filePath}&inJupyterLab=true&user=${user}`;
+    // wait for session to load
+  }
+
+  setOption(key: string, value: any): void {
+    // do nothing
+  }
+
+  setOptions(options: Partial<CodeEditor.IConfig>): void {
+    // do nothing
+  }
+
+  getCursorPosition(): CodeEditor.IPosition {
+    return {
+      line: 0,
+      column: 0
+    };
+  }
+
+  getLine(lineNumber: number): string | null {
+    return null;
+  }

```

Comment:
There is no constrain in JupyterLab about the API of the content widget. So I would highly suggest keeping it simple.